src/lib.rs

```diff
@@ -67,21 +82,77 @@ pub use poll::{Poll, Registry};
 pub use token::Token;
 pub use waker::Waker;
 
-#[cfg(unix)]
-cfg_any_os_util! {
-    pub mod unix {
-        //! Unix only extensions.
-        pub use crate::sys::SourceFd;
-    }
+#[cfg(all(unix, feature = "os-util"))]
+pub mod unix {
+    //! Unix only extensions.
+    pub use crate::sys::SourceFd;
+}
+
+// Enable with `cargo doc --features extra-docs`.
+#[cfg(feature = "extra-docs")]
+pub mod features {
+    //! # Mio's optional features.
+    //!
+    //! This document describes the available features in Mio.
+    //!
+    #![cfg_attr(feature = "os-poll", doc = "## `os-poll` (enabled)")]
+    #![cfg_attr(not(feature = "os-poll"), doc = "## `os-poll` (disabled)")]
+    //!
+    //! Mio by default provides only a shell implementation, that `panic!`s the
+    //! moment it is actually run. To run it requires OS support, this is
+    //! enabled by activating the `os-poll` feature.
+    //!
+    //! This makes `Poll`, `Registry` and `Waker` functional.
+    //!
+    #![cfg_attr(feature = "os-util", doc = "## `os-util` (enabled)")]
+    #![cfg_attr(not(feature = "os-util"), doc = "## `os-util` (disabled)")]
+    //!
+    //! `os-util` enables additional OS specific facilities. Currently this
+    //! means the `unix` module (with `SourceFd`) becomes available.
+    //!
+    //! ## Network types
+    //!
+    //! Mio provide three features to enable network types:
+    //!
+    #![cfg_attr(feature = "tcp", doc = "* `tcp` (enabled)")]
+    #![cfg_attr(not(feature = "tcp"), doc = "* `tcp` (disabled)")]
+    //! : includes `TcpStream` and `TcpListener`,
+    #![cfg_attr(feature = "udp", doc = "* `udp` (enabled)")]
+    #![cfg_attr(not(feature = "udp"), doc = "* `udp` (disabled)")]
+    //! : includes `UdpSocket`, and
+    #![cfg_attr(feature = "uds", doc = "* `uds` (enabled)")]
+    #![cfg_attr(not(feature = "uds"), doc = "* `uds` (disabled)")]
+    //! : includes `UnixDatagram`, `UnixListener`, `UnixStream` and `SocketAddr`.
```

Comment:
Could the net types here (`TcpStream`, `TcpListener`, `UdpSocket`, etc.) have links provided at the end of this doc section so that they are click throughs?

For example line 119 `//! : includes 'TcpStream' and 'TcpListener'` could be changed to `//! : includes ['TcpStream'] and ['TcpListener']`